unittests/chain_tests.cpp

```diff
@@ -74,4 +76,47 @@ BOOST_AUTO_TEST_CASE( replace_account_keys ) try {
 
 } FC_LOG_AND_RETHROW()
 
+BOOST_AUTO_TEST_CASE( decompressed_size_under_limit ) try {
+   tester chain;
+
+   // build a transaction, add cf data, sign
+   cf_action                        cfa;
+   eosio::chain::signed_transaction trx;
+   eosio::chain::action             act({}, cfa);
+   trx.context_free_actions.push_back(act);
+   // this is a under limit size  (4+4)*128*1024 = 1024*1024
+   for(int i = 0; i < 100*1024; ++i){
+      trx.context_free_data.emplace_back(fc::raw::pack<uint32_t>(100));
+      trx.context_free_data.emplace_back(fc::raw::pack<uint32_t>(200));
+   }
+   // add a normal action along with cfa
+   dummy_action         da = {DUMMY_ACTION_DEFAULT_A, DUMMY_ACTION_DEFAULT_B, DUMMY_ACTION_DEFAULT_C};
+   eosio::chain::action act1(
+       std::vector<eosio::chain::permission_level>{{"testapi"_n, eosio::chain::config::active_name}}, da);
+   trx.actions.push_back(act1);
+   chain.set_transaction_headers(trx);
+   auto sig = trx.sign(chain.get_private_key("testapi"_n, "active"), chain.control->get_chain_id());
+
+   // pack
+   packed_transaction ptv0(trx, packed_transaction::compression_type::zlib);
+   // try unpack and throw
+   bytes packed_txn = ptv0.get_packed_transaction();
+   bytes pcfd = ptv0.get_packed_context_free_data();
+   vector<signature_type>  sigs;
+   sigs.push_back(sig);
+   bool threw = false;
+   bool throw_expect_string = false;
+   try {
+      // will call into decompress zlib
+      packed_transaction copy( packed_txn, sigs, pcfd, packed_transaction::compression_type::zlib );
+   } catch(fc::exception& er){
+     threw = true;
+     if ( er.to_detail_string().find("Exceeded maximum decompressed transaction size") != std::string::npos){
+       throw_expect_string = true;
+     }
+   }
+   BOOST_REQUIRE(threw == false);
+   BOOST_REQUIRE(throw_expect_string == false);
```

Comment:
If we're trying to make sure it doesn't throw, which is how it is written, why do the try/catch at all? Unhandled exceptions will cause the test case to fail.

Reply:
I assume the try catch exists so we can search the exception for the failure we expect. I removed the variable and BOOST_REQUIRE to not throw at all.

Reply:
It's still not clear what the intent here is, and maybe @heifner can chime in since it seems like it's from his original PR?

The original code looked like it was trying to check that _nothing_ was thrown but it was doing so in a very roundabout way. This code looks like it allows _anything to be thrown that isn't "Exceeded maximum..."_ which would may or may not change the semantics.

I tried running the test locally and nothing was being thrown.

Reply:
I only side-ported it to develop-boxed. It appears it was written this odd way as an inverse of the test above it. My guess is the idea was that it is more obvious it is the opposite of the above test by just changing the BOOST_REQUIRE.

Reply:
I missed that, it explains it. But yeah that's not so pretty either. Should be something more on the order of
```c++
BOOST_REQUIRE_THROW(..., tx_decompression_error);
```
bonus points for
```c++
BOOST_REQUIRE_THROW(..., tx_decompression_error, [](const tx_decompression_error& e) {
      return e.to_detail_string().find("Exceeded maximum decompressed transaction size") != std::string::npos;
   }));
```

But at least its intent was not confusing like this one is.

Reply:
If we're going to touch the test (as we've started to), might as well clean it up further. If not, we can just PR changes to it later.

Reply:
I don't think `BOOST_REQUIRE_THROW` has a version which takes 3 arguments.

Reply:
yep that's the one